src/conn.c

```diff
@@ -4030,6 +4031,43 @@ natsConnection_GetLastError(natsConnection *nc, const char **lastError)
     return s;
 }
 
+natsStatus
+natsConnection_ReadLastError(natsConnection *nc, char *buf, size_t n)
+{
+    natsStatus  s;
+
+    if (nc == NULL)
+        return nats_setDefaultError(NATS_INVALID_ARG);
+
+    natsConn_Lock(nc);
+
+    s = nc->err;
+    if (s == NATS_OK)
+        nc->errStr[0] = '\0';
+    else if (nc->errStr[0] == '\0')
+        snprintf(nc->errStr, sizeof(nc->errStr), "%s", natsStatus_GetText(s));
+
+    if ((buf != NULL) && (n > 0))
+    {
+        size_t errLen = strlen(nc->errStr);
+        strncpy(buf, nc->errStr, n-1);
+        buf[n-1] = '\0';
+
+        bool truncate = ((errLen > (n-1)) && (n > 4));
+        if (truncate)
```

Comment:
But if you do, it would be nice to add a quick test with making sure we have a last error and then invoke with a smaller buffer and verify that this code is correct.